src/stdune/path.ml

```diff
@@ -156,6 +157,76 @@ end = struct
 
 end
 
+module Relative : sig
+  include Path_intf.S
+  val of_string : string -> (t, string) Result.t
+  val of_string_exn : string -> t
+  val relative : t -> string -> t
+  module L : sig
+    val relative : t -> string list -> t
+  end
+end = struct
+  type t = string
+  let of_string s =
+    if Filename.is_relative s then
+      Ok s
+    else
+      Error "path is not relative"
+  let of_string_exn s =
+    match of_string s with
+    | Ok s -> s
+    | Error msg ->
+      Exn.code_error "Relative.of_string_exn"
+        [ "s", Sexp.Encoder.string s
+        ; "msg", Sexp.Encoder.string msg
+        ]
+  let to_string s = s
+  let compare = String.compare
+  let hash = String.hash
+  let to_dyn s = Dyn.String s
+  let to_sexp s = Sexp.Encoder.string s
+  let pp = Format.pp_print_string
+  let is_root = function
+    | "" | "." -> true
+    | _ -> false
+  let parent_exn t =
+    let dirname = Filename.dirname t in
+    if dirname = "." then
+      Filename.concat t ".."
+    else
+      dirname
+  let basename s = Filename.basename s
+  let to_string_maybe_quoted s = String.maybe_quoted s
+  let extend_basename  t ~suffix =  t ^ suffix
```

Comment:
Depending on what we do with basename, `extend_basename` maybe should raise too.